### DISCUSSION TITLE
GMG and AMG

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#27125] [Q&A General] [answered]
In my research to determine the capability of MOOSE to perform Geometric Multi-Grid, it appears that the ability to apply GMG is limited. It appears the dominant method for use is AMG. Are there any developed examples in MOOSE that I can use to test GMG implementation?

**Comment by GiudGiud** [comment]
hello

I think we could do some form of GMG using adaptivity steps iirc. Start with a mesh that is coarse but that we want to refine, refine a few levels, then use coarsened solves to make progress faster.
@lindsayad do you know where to look for that?

Guillaume

**Reply by TLWise** [reply]
@GiudGiud Sounds like a great idea. Given my area of interest, it would appear too me that the lid-driven cavity flow problem would be an ideal development case.

**Reply by lindsayad** [reply, marked as answer]
We support grid sequencing in MOOSE, which is a loop outside of the the nonlinear solve (inside of the time stepper). You can grep for `num_grids` in the MOOSE repo for some demonstrations of that. However, that's clearly not GMG. We have not propagated GMG into MOOSE. The best place to read about GMG for libMesh is https://github.com/libMesh/libmesh/pull/2047
